File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [0] * n
        left, right = 0, n - 1

        for i in range(n - 1, -1, -1):
            left_num, right_num = abs(nums[left]) ,abs(nums[right])
            if right_num > left_num:
                ans[i] = pow(right_num, 2)
                right -= 1
            else:
                ans[i] = pow(left_num, 2)
                left += 1
        
        return ans

        

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        
        int n = nums.length;
        int[] answer = new int[n];
        int left = 0, right = n - 1;
        
        for (int i = n - 1; i >= 0; i--) {
            
            int square;
            
            // abs(left) > abs(right)
            if(Math.abs(nums[left]) < Math.abs(nums[right])) {
                square = (int) Math.pow(nums[right], 2);
                right --;
            } 
            // abs(left) <= abs(right)
            else {
                square = (int) Math.pow(nums[left], 2);
                left ++;
            }
            
            answer[i] = square;
        }
        
        return answer;
    }
}